envoy-control-core/src/test/kotlin/pl/allegro/tech/servicemesh/envoycontrol/utils/ReactorUtilsTest.kt

```diff
@@ -73,6 +73,33 @@ class ReactorUtilsTest {
         assertThat(buffered).isEqualTo(2.0)
     }
 
+    @Test
+    fun `should measure discarded items of backpressureLatest`() {
+        // given
+        val meterRegistry = SimpleMeterRegistry()
+
+        // when
+        val received = Flux.range(0, 10)
+            .filter { it % 2 == 0 } // 5 items will be discarded by filter
```

Comment:
Why do we need `filter` here?

Reply:
I believe it's about discarding difference :)

Reply:
There is comment on this line that I hoped explains it.
To be more clear: filter() triggers "onDiscard" event on the subscription. These discard will be counted in our metrics. So to correctly measure discarded event of given operator, we have to measure discarded items **before** and **after** and then calculate the difference. 

Filter simulates that situation and we test that discards caused be filter() are not counted in our final metric.

Let me know what should I write in the comment to explain it better.